daemon/api.go

```diff
@@ -421,6 +409,71 @@ func shouldSearchStore(r *http.Request) bool {
 	return false
 }
 
+// FIXME: add explicit test
+func storeUpdates(c *Command, r *http.Request) Response {
+	route := c.d.router.Get(snapCmd.Path)
+	if route == nil {
+		return InternalError("cannot find route for snaps")
+	}
+
+	found, err := allLocalSnapInfos(c.d.overlord.State())
+	if err != nil {
+		return InternalError("cannot list local snaps: %v", err)
+	}
+
+	candidatesInfo := make([]*store.RefreshCandidate, 0, len(found))
+	for _, sn := range found {
+		// snaps in try mode are not considered here
+		if sn.snapst.TryMode() {
+			continue
+		}
+
+		// get confinment preference from the snapstate
+		candidatesInfo = append(candidatesInfo, &store.RefreshCandidate{
+			// the desired channel (not sn.info.Channel!)
+			Channel: sn.snapst.Channel,
+			DevMode: sn.snapst.DevMode(),
+
+			SnapID:   sn.info.SnapID,
+			Revision: sn.info.Revision,
+			Epoch:    sn.info.Epoch,
+		})
+	}
+
+	auther, err := c.d.auther(r)
```

Comment:
searchStore should send the user in so we can do:
```
auther := user.Authenticator()
```
We should probably kill the auther helper to avoid redoing that historical use.